internal/net/http/middleware/timeout_test.go

```diff
@@ -128,160 +122,63 @@ func TestWrap(t *testing.T) {
 				},
 				field: field{
 					dur: 2 * time.Second,
-					eg:  errgroup.Get(),
 				},
 				checkFunc: func(code int, err error) error {
 					if !errors.Is(err, wantErr) {
 						return errors.Errorf("err not equals. want: %v, got: %v", wantErr, err)
 					}
-
 					if code != http.StatusInternalServerError {
 						return errors.Errorf("code is not equals. want: %v, got: %v", http.StatusInternalServerError, code)
 					}
-
-					if cnt != 1 {
+					if atomic.LoadInt32(&cnt) != 1 {
 						return errors.Errorf("called cnt is equals. want: %v, got: %v", 1, cnt)
 					}
-
 					return nil
 				},
 			}
 		}(),
-		func() test {
-			h := func(w http.ResponseWriter, req *http.Request) (code int, err error) {
-				time.Sleep(10 * time.Second)
-				return http.StatusOK, nil
-			}
-
-			return test{
-				name: "timeout processing of internally called handler",
-				args: args{
-					h: h,
-					w: new(httptest.ResponseRecorder),
-					r: new(http.Request),
-				},
-				field: field{
-					dur: 1 * time.Second,
-					eg:  errgroup.Get(),
-				},
-				checkFunc: func(code int, err error) error {
-					if err == nil {
-						return errors.Errorf("err is nil")
-					}
-
-					if code != http.StatusRequestTimeout {
-						return errors.Errorf("code is not equals. want: %v, got: %v", http.StatusRequestTimeout, code)
-					}
-
-					if !strings.Contains(err.Error(), "handler timeout") {
-						return errors.Errorf("err string no contains word of `handler timeout`")
-					}
-
-					return nil
+		{
+			name: "timeout processing of internally called handler",
+			args: args{
+				h: func(w http.ResponseWriter, req *http.Request) (code int, err error) {
+					return http.StatusOK, nil
 				},
-			}
-		}(),
+				w: new(httptest.ResponseRecorder),
+				r: new(http.Request),
+			},
+			field: field{
+				dur: 1, // set to extermemly small value to let it timeout
+			},
+			checkFunc: func(code int, err error) error {
+				if err == nil {
+					return errors.Errorf("err is nil")
+				}
+				if code != http.StatusRequestTimeout {
+					return errors.Errorf("code is not equals. want: %v, got: %v", http.StatusRequestTimeout, code)
+				}
+				if !strings.Contains(err.Error(), "handler timeout") {
+					return errors.Errorf("err string no contains word of `handler timeout`")
+				}
+
+				return nil
+			},
+		},
 	}
 
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
+	for _, tc := range tests {
+		test := tc
+		t.Run(test.name, func(tt *testing.T) {
+			tt.Parallel()
+
+			eg, _ := errgroup.New(context.Background())
 			to := &timeout{
-				dur: tt.field.dur,
-				eg:  tt.field.eg,
+				dur: test.field.dur,
+				eg:  eg,
 			}
-
-			code, err := to.Wrap(tt.args.h)(tt.args.w, tt.args.r)
-			if err := tt.checkFunc(code, err); err != nil {
+			code, err := to.Wrap(test.args.h)(test.args.w, test.args.r)
+			if err := test.checkFunc(code, err); err != nil {
 				t.Error(err)
 			}
 		})
 	}
 }
-
-func Test_timeout_Wrap(t *testing.T) {
```

Comment:
I am sorry that I forgot to mention it on PR.
The test function `TestWrap` is renamed to `Test_timeout_Wrap`, 
https://github.com/vdaas/vald/blob/main/internal/net/http/middleware/timeout_test.go#L56

since there is no `Wrap()` function is implemented in timeout.go (only *timeout.Wrap() is implemented) , I renamed this test function.
https://github.com/vdaas/vald/blob/main/internal/net/http/middleware/timeout.go#L46

Reply:
Got it